--- build.pl	2013-02-16 20:20:26.000000000 +0100
+++ build-test.pl	2013-02-22 13:36:18.000000000 +0100
@@ -657,9 +657,9 @@
 			#
 			foreach ($results->documentElement->getElementsByTagName("a")) {
 				my $href = $_->getAttribute("href");
-				if ($href =~ /^http:\/\/www.fsfe.org/) {
+				if ($href =~ /^http:\/\/test.fsfe.org/) {
 					if ($_->textContent != "Our global work") {
-						$href =~ s/http:\/\/www.fsfe.org//;
+						$href =~ s/http:\/\/test.fsfe.org//;
 					}
 				}
 				if (($href !~ /^http/) && ($href !~ /^#/)) {
